apps/storeutl.c

```diff
@@ -201,9 +201,7 @@ int storeutl_main(int argc, char *argv[])
             }
             break;
         case OPT_CRITERION_FINGERPRINT:
-            if (criterion != 0
-                || (criterion == OSSL_STORE_SEARCH_BY_KEY_FINGERPRINT
-                    && fingerprint != NULL)) {
+            if (criterion != 0) {
```

Comment:
Hold on!  What happened with `&& fingerprint != NULL`?

Reply:
It is checked just below on line 210.

Reply:
Hang on, is `OSSL_STORE_SEARCH_BY_KEY_FINGERPRINT` meant to be `0`? As long as `OSSL_STORE_SEARCH_BY_KEY_FINGERPRINT != 0`, then if `criterion == OSSL_STORE_SEARCH_BY_KEY_FINGERPRINT` it will make `criterion != 0` true, and we'll enter the block